modules/099-ceph-csi/hooks/get_crds.go

```diff
@@ -0,0 +1,197 @@
+}
+
+type Spec struct {
+	ClusterID string   `json:"clusterID"`
+	UserID    string   `json:"userID"`
+	UserKey   string   `json:"userKey"`
+	Monitors  []string `json:"monitors"`
+	Rbd       RBD      `json:"rbd,omitempty"`
+	CephFS    CephFS   `json:"cephfs,omitempty"`
+}
+
+type RBD struct {
+	StorageClasses []StorageClass `json:"storageClasses,omitempty"`
+}
+
+type CephFS struct {
+	StorageClasses []StorageClass `json:"storageClasses,omitempty"`
+	SubvolumeGroup string         `json:"subvolumeGroup,omitempty"`
+}
+
+type StorageClass struct {
+	NamePostfix          string   `json:"namePostfix"`
+	Pool                 string   `json:"pool,omitempty"`
+	ReclaimPolicy        string   `json:"reclaimPolicy,omitempty"`
+	AllowVolumeExpansion bool     `json:"allowVolumeExpansion,omitempty"`
+	MountOptions         []string `json:"mountOptions,omitempty"`
+	DefaultFSType        string   `json:"defaultFSType,omitempty"`
+	FsName               string   `json:"fsName,omitempty"`
+}
+
+type InternalValues struct {
+	Name string `json:"name"`
+	Spec Spec   `json:"spec"`
+}
+
+type CSIConfig struct {
+	ClusterID string          `json:"clusterID"`
+	Monitors  []string        `json:"monitors"`
+	CephFS    CSIConfigCephFS `json:"cephFS,omitempty"`
+}
+
+type CSIConfigCephFS struct {
+	SubvolumeGroup string `json:"subvolumeGroup,omitempty"`
+}
+
+type StorageClassFilter struct {
+	Name          string `json:"namePostfix"`
+	ReclaimPolicy string `json:"reclaimPolicy"`
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
+	Queue:        "/modules/ceph-csi",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "crs",
+			ApiVersion: "deckhouse.io/v1alpha1",
+			Kind:       "CephCSIDriver",
+			FilterFunc: applyCephCSIDriverFilter,
+		},
+		{
+			Name:       "scs",
+			ApiVersion: "storage.k8s.io/v1",
+			Kind:       "Storageclass",
+			LabelSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "ceph-csi",
+				},
+			},
+			FilterFunc: applyStorageclassFilter,
+		},
+	},
+}, dependency.WithExternalDependencies(setInternalValues))
+
+func applyCephCSIDriverFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var csi = &CephCSI{}
+	err := sdk.FromUnstructured(obj, csi)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert kubernetes object: %v", err)
+	}
+
+	return csi, nil
+}
+
+func applyStorageclassFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var sc = &storagev1.StorageClass{}
+	err := sdk.FromUnstructured(obj, sc)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert kubernetes object: %v", err)
+	}
+
+	return StorageClassFilter{
+		Name:          sc.Name,
+		ReclaimPolicy: string(*sc.ReclaimPolicy),
+	}, nil
+}
+
+func setInternalValues(input *go_hook.HookInput, dc dependency.Container) error {
+	values := []InternalValues{}
+	csiConfig := []CSIConfig{}
+
+	kubeClient, err := dc.GetK8sClient()
```

Comment:
You should avoid using kubeclient if you don't need it. In your case, it's better to stick to the patches and patch collector.

Reply:
I want to be sure that by the time helm starts, the storage class will be removed.

Reply:
Then it is ok to use patch collector.

Reply:
Replaced kubeclient with patchCollector.